src/main/java/bio/terra/workspace/service/resource/controlled/gcp/ControlledGcpResourceService.java

```diff
@@ -37,4 +46,16 @@ public String createGcsBucket(
 
     return jobBuilder.submit();
   }
+
+  /**
+   * Retrieve the bucket metadata from the DAO(s)
+   * @param workspaceId - workspace containing this bucket
+   * @param resourceId - resource ID for the bucket
+   * @return attributes we store about this bucket (i.e. its name)
+   */
+  public GoogleBucketStoredAttributes getBucket(UUID workspaceId, UUID resourceId) {
```

Comment:
This needs an authz check like `validateWorkspaceAndAction` from workspace service, otherwise it's open to everyone instead of just workspace readers.

Reply:
Yeah, spoke with @ddietterich a bit on that. I got the impression we were waiting on something to do this.

```
We do need to check permissions.
10:01
But, we still need the new workspace model to do it right. Sigh…
10:02
That is, the new Sam model for workspace permissions that Zach is doing
```

Reply:
Ah good point - we should really check with Sam whether the user has read permission on the specified resource, not general read access to the workspace. Can you add a todo pointing to PF-413?